packages/wrangler/src/api/startDevWorker/BundlerController.ts

```diff
@@ -247,17 +254,22 @@ export class BundlerController extends Controller<BundlerControllerEventMap> {
 			},
 			(cb) => {
 				const newBundle = cb(this.#currentBundle);
-				this.emitBundleCompleteEvent(config, newBundle);
-				this.#currentBundle = newBundle;
+				if (!buildAborter.signal.aborted) {
+					this.emitBundleCompleteEvent(config, newBundle);
+					this.#currentBundle = newBundle;
+				}
 			},
-			(err) =>
-				this.emitErrorEvent({
-					type: "error",
-					reason: "Failed to construct initial bundle",
-					cause: castErrorCause(err),
-					source: "BundlerController",
-					data: undefined,
-				})
+			(err) => {
+				if (!buildAborter.signal.aborted) {
+					this.emitErrorEvent({
+						type: "error",
+						reason: "Failed to construct initial bundle",
+						cause: castErrorCause(err),
+						source: "BundlerController",
+						data: undefined,
+					});
+				}
+			}
```

Comment:
BundleController had a race condition where old builds would sometimes overwrite newer ones. We'd thought this was handled by esbuild, but it turns out that it's not in the way we need it to. This fixes the intermittent "Failed to construct initial bundle" flakes we were seeing in E2E tests in CI

Reply:
Very cool, thanks for that.